cmd/worker/main.go

```diff
@@ -6,45 +6,63 @@ import (
 	"os"
 	"time"
 
+	"github.com/google/uuid"
 	"github.com/phuslu/log"
 
 	"github.com/hrand1005/ZoneScaler/common"
 	"github.com/hrand1005/ZoneScaler/worker"
 )
 
+const (
+	maxRetries    = 30
+	retryInterval = 1 * time.Second
+)
+
 func main() {
 	if len(os.Args) < 2 {
 		log.Fatal().Msg("Usage: ./coordinator <JSON config>")
+		return
 	}
 
 	conf, err := worker.LoadConfig(os.Args[1])
 	if err != nil {
 		log.Fatal().Err(err).Msgf("Failed to load config")
+		return
 	}
 
 	node := common.GameNode{
-		ID:            "unique-node-id",
+		ID:            uuid.NewString(),
 		Address:       fmt.Sprintf("%s:%d", conf.Host, conf.Port),
 		Load:          0,
 		Regions:       []string{"region1", "region2"},
 		IsActive:      true,
 		LastHeartbeat: time.Now(),
 	}
 
-	coordinatorAddr := fmt.Sprintf("%v:%v", conf.CoordinatorHost, conf.CoordinatorPort)
-	err = worker.RegisterWithCoordinator(coordinatorAddr, node)
-	if err != nil {
-		log.Fatal().Err(err)
+	coordinatorAddr := fmt.Sprintf("%v:%v", os.Getenv("COORDINATOR_HOST"), os.Getenv("COORDINATOR_PORT"))
+
+	registered := false
+	for try := 0; try < maxRetries; try++ {
```

Comment:
I threw this in because I didn't know if there would be a race with docker-compose. Even though there's a 'depends_on' field for the worker service, I think that only applies to the building of the docker image -- no guarantees can be made about the state of the coordinator service when the worker reaches this point.

Reply:
It can be done, depends_on is used in building and starting up until healthy.
This "until healthy" is the key, as by default it just means it started running the code, which would lead into the race you're suggesting. But we can customize it: look into this example: https://github.com/peter-evans/docker-compose-healthcheck/blob/master/docker-compose.yml or this tutorial: https://medium.com/@saklani1408/configuring-healthcheck-in-docker-compose-3fa6439ee280

Reply:
To do this, it looks like we'd want to implement some "coordinator_is_ready" script/utility in order to be able to do this line: 
https://github.com/peter-evans/docker-compose-healthcheck/blob/master/docker-compose.yml#L12

Would you prefer that to retries in Go? I don't feel strongly either way.

Reply:
I forgot to reply to this, no need to implement a script, you can just curl an endpoint in the service for a healthcheck